sql/src/main/java/io/crate/operation/collect/MapSideDataCollectOperation.java

```diff
@@ -146,8 +146,15 @@ public ResultProvider createDownstream(CollectNode node) {
         if (ExecutionNodes.hasDirectResponseDownstream(node.downstreamNodes())) {
             return new SingleBucketBuilder(getStreamers(node));
         } else {
+            // TODO: set bucketIdx properly
+            ArrayList<String> server = Lists.newArrayList(node.executionNodes());
+            Collections.sort(server);
```

Comment:
executionNodes is a Set, so the sort is required but if we find a better solution to set the bucketIdx it would be nice =)